src/scripts/main.js

```diff
@@ -1,3 +1,258 @@
+
+    document.getElementById('board').innerHTML = '';
+    score.textContent = '0';
+    scoreSum = 0;
+    setGame();
+  } else if (gameOn === false) {
+    startGame.classList.remove('restart');
+    startGame.classList.add('start');
+    startGame.innerHTML = 'Start';
+  }
+}
+
+startGame.addEventListener('click', e => {
+  e.preventDefault();
+
+  setGame();
+  gameOn = true;
+  messageLose.classList.add('hidden');
+  messageWin.classList.add('hidden');
+  messageStart.classList.add('hidden');
+  messageInstruction.classList.add('hidden');
+  buttonSwitcher();
+});
+
+function setGame() {
+  board = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < colums; c++) {
+      const tile = document.createElement('div');
+
+      tile.id = r.toString() + '-' + c.toString();
+
+      const num = board[r][c];
+
+      updateTile(tile, num);
+
+      document.getElementById('board').append(tile);
+    }
+  }
+
+  setRandom();
+  setRandom();
+}
+
+function hasEmptyTile() {
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < colums; c++) {
+      if (board[r][c] === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function setRandom() {
+  if (!hasEmptyTile()) {
+    messageLose.classList.remove('hidden');
+    gameOn = false;
+    document.removeEventListener('keyup', keyUpHandler);
+
+    return;
+  }
+
+  let found = false;
+
+  while (!found) {
+    const r = Math.floor(Math.random() * rows);
+    const c = Math.floor(Math.random() * colums);
+
+    if (board[r][c] === 0) {
+      const random = Math.random() < 0.9 ? 2 : 4;
+
+      board[r][c] = random;
+
+      const tile = document.getElementById(r.toString() + '-' + c.toString());
+
+      tile.textContent = random;
+      tile.classList.add(`${random}`);
+
+      found = true;
+    }
+  }
+}
+
+function updateTile(tile, num) {
+  tile.innerText = '';
+  tile.classList.value = '';
+  tile.classList.add('tile');
+
+  if (num > 0) {
+    tile.innerText = num;
+
+    if (num < 2048) {
+      tile.classList.add('tile--' + num.toString());
+    }
+
+    if (num === 2048) {
+      tile.classList.add('tile--' + num.toString());
+      messageWin.classList.remove('hidden');
+      document.removeEventListener('keyup', keyUpHandler);
+    }
+  }
+}
+
+function keyUpHandler(e) {
+  currnetScore = 0;
+
+  if (e.code === 'ArrowLeft') {
+    slideLeft();
+    setRandom();
+  } else if (e.code === 'ArrowRight') {
+    slideRight();
+    setRandom();
+  } else if (e.code === 'ArrowUp') {
+    slideUp();
+    setRandom();
+  } else if (e.code === 'ArrowDown') {
+    slideDown();
+    setRandom();
+  }
+  score.textContent = scoreSum;
+
+  score.insertAdjacentHTML('afterend', `
+    <span class="score-update">+${currnetScore}</span>
+  `);
+}
+
+if (document !== 'undefined') {
+  document.addEventListener('keyup', keyUpHandler, (e) => {
+    keyUpHandler();
+  });
```

Comment:
Thirdt params for addEventListener - should be an options. [Read more](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#syntax)
```suggestion
  document.addEventListener('keyup', keyUpHandler);
```